sshuttle/bin/sudoers-add

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+# William Mantly <wmantly@gmail.com>
+# MIT License
+# https://github.com/wmantly/sudoers-add
+
+NEWLINE=$'\n'
+CONTENT=""
+ME="$(basename "$(test -L "$0" && readlink "$0" || echo "$0")")"
+
+if [ "$1" == "--help" ] || [ "$1" == "-h" ]; then
+	echo "Usage: $ME [file_path] [sudoers-file-name]"
+	echo "Usage: [content] | $ME sudoers-file-name"
+	echo "This will take a sudoers config validate it and add it to /etc/sudoers.d/{sudoers-file-name}"
+	echo "The config can come from a file, first usage example or piped in second example."
+	exit 0
+fi
+
+if [ "$1" == "" ]; then
+	(>&2 echo "This command take at lest one argument. See $ME --help")
+	exit 1	
+fi
+
+if [ "$2" == "" ]; then
+	FILE_NAME=$1
+	shift
+else
+	FILE_NAME=$2
+fi
+
+if [[ $EUID -ne 0 ]]; then
+	echo "This script must be run as root" 
+	exit 1
+fi
+
+while read -r line
+do
+	CONTENT+="${line}${NEWLINE}"
+done < "${1:-/dev/stdin}"
+
+if [ "$CONTENT" == "" ]; then
+	(>&2 echo "No config content specified. See $ME --help")
+	exit 1
+fi
+
+if [ "$FILE_NAME" == "" ]; then
+	(>&2 echo "No sudoers file name specified. See $ME --help")
+	exit 1
+fi
+
+touch "/tmp/$FILE_NAME"
```

Comment:
This (and next lines) seems unsafe use of `/tmp`. See https://www.netmeister.org/blog/mktemp.html

Reply:
Additionally to the important requirement not to use /tmp, i have a small suggestion:

After a successful visudo test on the generated sudoers snippet you should not move the temp sudoers-file, you should rather write the `$CONTENT` to the final destination `/etc/sudoers.d/$FILE_NAME`. This way you can avoid problems, in case the temporary file changed without intent, or was modified with malicious intent.

A possible rewrite with the above in mind:

```
umask 077
checksudoers=$(mktemp sshuttle-sudoers-rule.XXXXXXXXX)
[ -z "${checksudoers}" ] && echo "$CONTENT" > "${checksudoers}"

if [ visudo -c -q -f "${checksudoers}" ]; then
	if [ -e "/etc/sudoers.d/$FILE_NAME" ]; then
		rm "/etc/sudoers.d/$FILE_NAME"
	fi
	echo "$CONTENT" > "/etc/sudoers.d/$FILE_NAME"
	chmod 0440 "/etc/sudoers.d/$FILE_NAME"
	rm "${checksudoers}"
	exit 0
else
	visudo -c -f "${checksudoers}"
	rm "${checksudoers}"
	exit 1
fi
```

Reply:
I will change this.